File: AdventOfCode2020/day05.py
def read_boarding_passes( filename ):

    with open(filename) as f:
        tickets = f.read()
        table = str.maketrans(dict(zip("FBLR","0101")))
        tickets = tickets.translate(table)
        return tickets.split()

tickets = read_boarding_passes( "input05.txt")
seat_ids = list(map( lambda x: int(x, base=2), tickets))
print(max(seat_ids))
print(set(range(min(seat_ids),max(seat_ids))) - set(seat_ids))

File: AdventOfCode2020/day05.jl
function read_boarding_passes(filename)

    return open(filename) do f
        return readlines(f)
    end

end

tickets = read_boarding_passes("input05.txt")
tickets .= replace.(tickets, r"[RB]" => "1")
tickets .= replace.(tickets, r"[FL]" => "0")
seat_id = parse.(Int, tickets, base = 2)

println(maximum(seat_id))
seat = setdiff(minimum(seat_id):maximum(seat_id), seat_id)
println(seat)
